src/main/java/com/solvd/webtesting/page/ResultPage.java

```diff
@@ -0,0 +1,29 @@
+package com.solvd.webtesting.page;
+
+import com.qaprosoft.carina.core.foundation.webdriver.decorator.ExtendedWebElement;
+import com.qaprosoft.carina.core.gui.AbstractPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ResultPage extends AbstractPage {
+    @FindBy(css = ".result__name")
+    private List<ExtendedWebElement> searchResultItems;
+
+    public ResultPage(WebDriver driver) {
+        super(driver);
+        setPageURL("");
+    }
+
+    public List<String> getFullNames() {
+        return searchResultItems.stream()
+                .map(ExtendedWebElement::getText)
+                .collect(Collectors.toList());
+    }
+
+    public List<ExtendedWebElement> getSearchResultItems() {
+        return searchResultItems;
+    }
```

Comment:
The same: Please don't return ExtendedWebElement, if use need to do some manipulations with elements, u need to do this on page where it present.